x-pack/legacy/plugins/maps/public/actions/map_actions.js

```diff
@@ -615,7 +615,19 @@ export function removeSelectedLayer() {
   };
 }
 
-export function removeLayer(layerId) {
+export function safeRemoveLayer(layerId) {
```

Comment:
I think a more descriptive name then `safeRemoveLayer` should be used since the function is exported. What does `safe` mean? 

The existing `removeLayer` function is only used internally to map_actions. How about renaming that function to `removeLayerFromList`.

Then just name the exported function to `removeLayer`.

I think `removeSelectedLayer` would be no longer needed and it could just use the new `removeLayer` function